packages/core/src/Services/socket-general.js

```diff
@@ -307,21 +315,25 @@ export default BinarySocketGeneral;
 const ResponseHandlers = (() => {
     const websiteStatus = response => {
         if (response.website_status) {
-            const is_available = !BinarySocket.isSiteDown(response.website_status.site_status);
-            if (is_available && BinarySocket.getAvailability().is_down) {
-                window.location.reload();
-                return;
-            }
-            const is_updating = BinarySocket.isSiteUpdating(response.website_status.site_status);
-            if (is_updating && !BinarySocket.getAvailability().is_updating) {
-                // the existing connection is alive for one minute while status is updating
-                // switch to the new connection somewhere between 1-30 seconds from now
-                // to avoid everyone switching to the new connection at the same time
-                const rand_timeout = Math.floor(Math.random() * 30) + 1;
+            const is_server_down = checkServerMaintenance(response.website_status);
+
+            // If site is down or updating, connect to WebSocket with an exponentially increasing delay on every attempt.
+            // Starts off with 1.024 seconds and grow exponentially.
+            if (is_server_down) {
+                const reconnectionDelay = Math.min(Math.pow(2, reconnectionCounter + 9), 600000);
```

Comment:
Should we not also include a random element here? 

Maybe something like:
const reconnectionDelay = Math.min(Math.pow(2, reconnectionCounter + 9), 600000)*((Math.random()+9.5)/10);

I am worried that if all of the clients break at the same time they will all follow the same backoff, and all swamp us at the same time when they try to reconnect.